shared/src/hooks/pendingTransactionsContext.tsx

```diff
@@ -45,15 +50,37 @@ export const PendingTransactionsContextProvider: React.FC<{
     "pendingTransactions"
   );
   const [transactionsCounter, setTransactionsCounter] = useState(0);
-  const { library } = useWeb3React();
+  const { ethereumProvider } = useWeb3Wallet();
+  const [chain] = useChain();
+  const { connection } = useConnection();
 
   /**
    * Keep track with first confirmation
    */
   useEffect(() => {
     pendingTransactions.forEach(async (transaction) => {
       if (!transaction.status) {
-        const receipt = await library.waitForTransaction(transaction.txhash, 5);
+        let receipt: any;
+
+        if (isEVMChain(chain)) {
+          receipt = await (
+            ethereumProvider as providers.Web3Provider
+          ).waitForTransaction(transaction.txhash, 5);
+        } else if (isSolanaChain(chain)) {
+          receipt = await connection.confirmTransaction(
+            transaction.txhash,
+            "confirmed"
+          );
+        }
+
+        const getTransactionStatus = () => {
```

Comment:
Do you think if we extract this out into utils would be better? Since we will be checking the status in multiple places